cachi2/core/models/sbom.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Literal, Optional
+
+import pydantic
+
+from cachi2.core.models.validators import unique_sorted
+
+PropertyName = Literal[
+    "cachi2:found_by",
+    "cachi2:missing_hash:in_file",
+    "cdx:npm:package:bundled",
+    "cdx:npm:package:development",
+]
+
+HashAlgorithm = Literal["SHA-256", "SHA-384", "SHA-512"]
+
+
+class Hash(pydantic.BaseModel, frozen=True):
+    """A checksum related to an SBOM component."""
+
+    algorithm: HashAlgorithm
+    content: str
+
+
+class Property(pydantic.BaseModel):
+    """A property inside an SBOM component."""
+
+    name: PropertyName
+    value: str
+
+
+FOUND_BY_CACHI2_PROPERTY: Property = Property(name="cachi2:found_by", value="cachi2")
+
+
+class Component(pydantic.BaseModel):
+    """A software component such as a dependency or a package.
+
+    Compliant to the CycloneDX specification:
+    https://cyclonedx.org/docs/1.4/json/#components
+    """
+
+    name: str
+    purl: str
+    version: Optional[str] = None
+    properties: list[Property] = []
+    type: Literal["library"] = "library"
+    hashes: list[Hash] = []
```

Comment:
Having a default empty list here makes the merging easier, since we can always transform this attribute into a frozenset. But the downside is that the SBOM components will always have a `hash` attribute, even if it's an empty list =/

Reply:
And we don't really need to report hashes yet, the missing_hash property works with or without that

By adding it now we force the npm and gomod implementations to report hashes as well

Reply:
Great, this makes things much easier :)